### ISSUE TITLE
macOS + home-manager: device_id is ~ in client.yml

### ISSUE THREAD

**Issue by rpearce** [Rigellute/spotify-tui#653]
<img width="988" alt="image" src="https://user-images.githubusercontent.com/592876/98446278-4ffd6080-2114-11eb-880a-5e85c789e066.png">

Hi, thank you for this project! Amazing.

I'm using macOS with nix+home-manager, and I've got everything running smoothly and have Spotify Premium, but when I try to play a song, it can't figure out what device to use. I'm on a MacBook Pro.

Here's my `client.yml`:

```
---
client_id: ...
client_secret: ...
device_id: ~
port: 8888
```

It seems like that `device_id` isn't right? How can I fix that?

Thanks for your help!

**Comment by andersfylling** [comment]
Exactly the same issue. I'm on arch linux. Though I am not running a the official spotify application on my computer (I assumed spotifyd was a replacement from their README).

**Comment by schwittlick** [comment]
same issue.. it only works when the spotify app is running in parallel :/

**Comment by andersfylling** [comment]
If the spotifyd is running in the background, the music has to be started from an official spotify client via Spotify Connect for me. I had some serious issue getting spotifyd working in the first place.

**Comment by Rigellute** [comment]
Hi! This sounds like an issue with `spotifyd`. The device list in `spotify-tui` is the result of an API call to Spotify. Sounds like `spotifyd` is not discoverable for some reason?

**Comment by andersfylling** [comment]
Yeah I believe so as well, seems like it can't "discover itself", until "triggered" by another device through Spotify Connect. If that makes sense.

Spotifyd is able to see other spotify devices/instances, just not itself.

**Comment by codygman** [comment]
When I get this issue just restaring spotifyd with `systemctl --user restart spotifyd.service` fixes the issue.

**Comment by codygman** [comment]
https://github.com/Spotifyd/spotifyd/pull/596
